### PR TITLE
order-history

### PR DESCRIPTION
**Pull request by koloale**

### CODE REVIEW

File: src/main/scala/com/wavesplatform/matcher/model/OrderHistory.scala [hunk 1]
```diff
@@ -1,101 +1,127 @@
...
-  val ordersRemainingAmount: Cache[String, (Long, Long)] =
-    TTLCache[String, (Long, Long)]((Order.MaxLiveTime + 3600*1000).millis)
-  val openOrdersCount = mutable.Map.empty[AddressString, Int]
+  def didOrderAccepted(event: OrderAdded): Unit
+  def didOrderExecuted(event: OrderExecuted): Unit
+  def didOrderCanceled(event: OrderCanceled)
+  def getOrderStatus(id: String): OrderStatus
+  def getOrderInfo(id: String): OrderInfo
+  def getOpenVolume(assetAcc: AssetAcc): Long
+  def getOrdersByPairAndAddress(assetPair: AssetPair, address: String): Set[String]
```
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/model/OrderHistory.scala]: done

File: src/main/scala/com/wavesplatform/matcher/model/OrderHistory.scala [hunk 2]
```diff
@@ -1,101 +1,127 @@
...
 trait OrderHistory {
-  type AddressString = String
-
-  val assetsToSpend = mutable.Map.empty[AddressString, Long]
-  val ordersRemainingAmount: Cache[String, (Long, Long)] =
-    TTLCache[String, (Long, Long)]((Order.MaxLiveTime + 3600*1000).millis)
-  val openOrdersCount = mutable.Map.empty[AddressString, Int]
+  def didOrderAccepted(event: OrderAdded): Unit
+  def didOrderExecuted(event: OrderExecuted): Unit
+  def didOrderCanceled(event: OrderCanceled)
```
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/model/OrderHistory.scala]: done

File: src/main/scala/com/wavesplatform/matcher/api/MatcherApiRoute.scala [hunk 3]
```diff
@@ -122,6 +127,93 @@ case class MatcherApiRoute(wallet: Wallet,storedState: StateReader, matcher: Act
...
+            .mapTo[MatcherResponse]
+            .map(r => r.code -> r.json)
+        } else {
+          StatusCodes.BadRequest -> Json.obj("message" -> "Incorrect signature")
+        }
+      }
+    }
+  }
+
+  def checkGetSignature(publicKey: String, timestamp: String, signature: String): Try[String] = {
```
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/api/MatcherApiRoute.scala]: agree. done.

File: src/main/resources/application.conf [hunk 4]
```diff
@@ -258,14 +258,15 @@ waves {
...
-    min-order-fee = 100000
+    min-order-fee = 300000
 
     # Fee of order match transaction
     order-match-tx-fee = 300000
 
     # Matcher's directories
     journal-directory = ${waves.directory}"/journal"
     snapshots-directory = ${waves.directory}"/snapshots"
+    order-history-file = ${waves.directory}"/matcher/orderhistory.dat"
```
Comment (koloale) [inline src/main/resources/application.conf]: done.

File: src/main/resources/application.conf [hunk 5]
```diff
@@ -258,14 +258,15 @@ waves {
...
-    min-order-fee = 100000
+    min-order-fee = 300000
 
     # Fee of order match transaction
     order-match-tx-fee = 300000
 
     # Matcher's directories
-    journal-directory = ${waves.directory}"/journal"
-    snapshots-directory = ${waves.directory}"/snapshots"
+    journal-directory = ${waves.directory}"/matcher/journal"
```
Comment (koloale) [inline src/main/resources/application.conf]: done

File: src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala [hunk 6]
```diff
@@ -18,7 +20,10 @@ case class MatcherSettings(enable: Boolean,
                            snapshotsInterval: FiniteDuration,
                            maxOpenOrders: Int,
                            priceAssets: Seq[String],
-                           predefinedPairs: Seq[AssetPair]
+                           predefinedPairs: Seq[AssetPair],
+                           maxTimestampDiff: FiniteDuration,
+                           orderHistoryFile: String,
+                           isMigrateToNewOrderHistoryStorage: Boolean
```
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala]: It's migration from the previous storage in the snapshot to MVStore. It should be done once. Or can be done manually if we need to rebuild history. The settings is enough here

Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala]: Ok made this settings calculable

Comment (ismagin) [inline src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala]: It's good that it's calculable, but it still shouldn't be here. It is not a "setting" of any kind.

**Pull request merged**